### ISSUE TITLE
FromForm and ToForm instances for (,) and Either

### ISSUE THREAD

**Issue by wraithm** [fizruk/http-api-data#91]
Does it make sense to have:

```
instance (FromForm a, FromForm b) => FromForm (a, b) where
    fromFrom f = (,) <$> fromForm f <*> fromForm f
```

and ditto for Either?

**Comment by fizruk** [comment]
@wraithm not sure, do you have a use case in mind?

**Comment by wraithm** [comment]
The specific example is that I have a form for creating groups of users, and then I have a form for creating users. For certain kinds of groups, I need a form with a "root" user for that group, so I just use a tuple of the basic group form and the new user form for those kinds of group forms.

So, I have two forms on two separate pages, and then I have a third form that is basically just both of those two forms on the same form. Of course, I could make a new data type with a new instance, but it was very convenient to use a tuple, and I was surprised to find that this instance didn't already exist. Also, you could have an Either version of this as well, but the tuple case is what I used for my specific problem here.

**Comment by phadej** [comment]
Question: can corresponding `ToForm` be written? If it can, then I don't see downsides.

**Comment by wraithm** [comment]
Yes, you can definitely write a ToForm instance for both (,) and Either.
The semantics of toForm after fromForm and vice versa would maybe be a
little weird, but I don't think that matters very much.

**Comment by phadej** [comment]
I think we can assume (and mention that in instance haddocks) that forms "read/write" disjoint field sets.
